src/cli/index.js

```diff
@@ -172,10 +172,17 @@ export function main({
   const command = commands[commandName];
 
   let warnAboutRunDashDash = false;
-  // we are using "yarn <script> -abc" or "yarn run <script> -abc", we want -abc to be script options, not yarn options
-  if (command === commands.run || command === commands.create) {
+  // we are using "yarn <script> -abc", "yarn run <script> -abc", or "yarn node -abc", we want -abc to
+  // be script options, not yarn options
+  if (command === commands.run || command === commands.create || command === commands.node) {
```

Comment:
Looks like it is time to create `ARG_PROXY_COMMANDS = new Set('run', 'create', 'node');` or add a new field to all commands, `proxyArgs: boolean` to determine this behavior.